### ISSUE TITLE
daemon: native TLS for RPC

### ISSUE THREAD

**Issue by ilf** [transmission/transmission#777]
Currently, the [RPC communication](https://github.com/transmission/transmission/blob/master/extras/rpc-spec.txt) is via cleartext HTTP. For confidentiality, integrity and authenticity, supporting Transport Layer Security (TLS) would be awesome. Currently, some users do this [via a reverse proxy like lighttpd](https://unix.stackexchange.com/questions/467483/getting-https-web-interface-for-transmission-with-lighttpd-reverse-proxy). Native support for this would be awesome, because it's cleaner, more accessible and using less ressources.

**Comment by gclawes** [comment]
The above guide appears to be only for the web interface.  It seems like the Qt client would also need support for SSL.

There was an old trac ticket [here](https://trac.transmissionbt.com/ticket/5412), but it hasn't been touched in years.

**Comment by gclawes** [comment]
Some linux distros apparently provide a patched version: https://aur.archlinux.org/packages/transmission-qt-ssl-git/

**Comment by ckerr** [comment]
> Some linux distros apparently provide a patched version: https://aur.archlinux.org/packages/transmission-qt-ssl-git/

From looking at this patch, it appears to be a settings option in transmission-qt to force 'https' for remote connections, but doesn't do anything on the server side to support an https server. Probably this patch is useful only in conjunction with running a proxy.

Transmission currently uses libevent's http server, which was handy because Transmission was already using libevent and because both Transmission and libevent were written in C.

Now that Transmission has migrated to C++, that makes several other options much easier to use. For example it might be reasonable to replace evhttp with something like https://github.com/yhirose/cpp-httplib or https://github.com/Stiffstream/restinio

**Comment by gclawes** [comment]
> From looking at this patch, it appears to be a settings option in transmission-qt to force 'https' for remote connections, but doesn't do anything on the server side to support an https server. Probably this patch is useful only in conjunction with running a proxy.

Yeah, it looks like that's the case.  Still, that's a good first step.  I don't think it's worth blocking the client implementation while waiting for sever-side support, given that running a reverse proxy with https is a totally valid server config.
